### ISSUE TITLE
Python3: TypeError: a bytes-like object is required, not 'str'

### ISSUE THREAD

**Issue by kingname** [rolando/scrapy-redis#70]
In the `spider.py` line 96:

```
if '://' in data:
```

In python 3, the data is bytes type, but '://' is a str, so here will raise error.

My solution is below, pay attention: only `data` is not None can it be decode()

``` python
def next_requests(self):
        """Returns a request to be scheduled or none."""
        use_set = self.settings.getbool('REDIS_START_URLS_AS_SET')
        fetch_one = self.server.spop if use_set else self.server.lpop
        # XXX: Do we need to use a timeout here?
        found = 0
        while found < self.redis_batch_size:
            data = fetch_one(self.redis_key)
            if not data:
                # Queue empty.
                break
            data = data.decode() # this line work fine in both Py2 and Py3
            req = self.make_request_from_data(data)
            if req:
                yield req
                found += 1
            else:
                self.logger.debug("Request not made from data: %r", data)

        if found:
            self.logger.debug("Read %s requests from '%s'", found, self.redis_key)
```

**Comment by rolando** [comment]
Thanks for your input. Here is a WIP PR: https://github.com/rolando/scrapy-redis/pull/71

**Comment by rolando** [comment]
Fixed in #71
